src/Controller/EsiController.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2016 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+
+namespace Contao\CoreBundle\Controller;
+
+use Contao\CoreBundle\Framework\ContaoFrameworkInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Handles the Contao ESI requests. This might be subject to change
+ * in the very near future which is why this class is declared final.
+ *
+ * @author Yanick Witschi <https://github.com/toflar>
+ */
+final class EsiController extends Controller
```

Comment:
Declaring a class `final` is not the appropriate thing to do if it is subject to change. We should rather declare it `@internal` instead, shouldn't we?